pdm/cli/commands/run.py

```diff
@@ -151,11 +152,15 @@ def _run_process(
                 process_env["NO_SITE_PACKAGES"] = "1"
 
         cwd = project.root if chdir else None
-        process = subprocess.Popen(expanded_args, cwd=cwd, env=process_env, shell=shell)
-        try:
-            process.wait()
-        except KeyboardInterrupt:
-            pass
+
+        s = signal.signal(
+            signal.SIGINT, lambda signum, frame: process.send_signal(signum)
+        )
+        process = subprocess.Popen(
+            expanded_args, cwd=cwd, env=process_env, shell=shell, bufsize=0
+        )
+        process.wait()
+        signal.signal(signal.SIGINT, s)
```

Comment:
Wouldn't it also work to just `process.send_signal(signum)` in the previous code's `except KeyboardInterrupt:` block? (without (un)installing a custom signal handler)

Reply:
This means the parent process will catch the interrupt and end the function, unless we `process.wait()` again